src/split.c

```diff
@@ -64,9 +64,7 @@ rcutils_split(
       ++string_array->size;
     }
   }
-  // TODO(wjwwood): refactor this function so it can use rcutils_string_array_init() instead
-  string_array->data = allocator.allocate(string_array->size * sizeof(char *), allocator.state);
-  if (NULL == string_array->data) {
+  if (rcutils_string_array_init(string_array, string_array->size, &allocator) != RCUTILS_RET_OK) {
```

Comment:
It's kind of weird to me that we setup `string_array->size`, and then pass it into `rcutils_string_array_init`, which sets it back to itself.  It is not wrong, per-se, but just reads strangely.

I might suggest that we use a stack variable to compute the size, and then pass that into `rcutils_string_array_init`.  That will end up with the same result, but read better in my opinion.

Reply:
Relatedly, we can remove the assignment of `string_array->allocator` at line 45, since `rcutils_string_array_init` will do that as well.

Reply:
done!